contrib/apparmor/template.go

```diff
@@ -53,6 +53,10 @@ profile {{.Name}} flags=(attach_disconnected,mediate_deleted) {
   umount,
   # Host (privileged) processes may send signals to container processes.
   signal (receive) peer=unconfined,
+  # runc may send signals to container processes.
+  signal (receive) peer=runc,
+  # crun may send signals to container processes.
+  signal (receive) peer=crun,
```

Comment:
Should this be set only when the corresponding profile exist?

Reply:
Also, what about youki, gvisor, kata, etc ?

Reply:
> Should this be set only when the corresponding profile exist?

I will update the template to check for known profiles.

> Also, what about youki, gvisor, kata, etc ?

These currently don't have any default profiles, see https://gitlab.com/apparmor/apparmor/-/tree/master/profiles/apparmor.d.

I don't know much about gVisor and kata runtimes, but I think those run containers in virtual machine, so their `kill` shouldn't involve sending signals.

Reply:
@AkihiroSuda What is the deadline for this change to be approved and merged to make it into v1.7.16?

I'm not sure what method would you recommend to conditionally add the signal receive rules. I can think of 3 ways:
1. Abuse `macroExists(profile)` to check if `/etc/apparmor.d/$profile` exists for each profile in a hardcoded list of OCI runtime profiles known to exist in latest AppArmor
2. Use `isLoaded(profile)` to check if the profile is loaded for each profile as above.
3. Bring back `parseVersion` that was removed here https://github.com/containerd/containerd/pull/8069 and conditionally add profiles based on known OCI runtime profiles in the parsed AppArmor version.
4. Keep it as it is, rely on the fact that it's regular expression and it won't match profiles that are not loaded. This is almost like option 2.
5. 
Which option would you go with or do you know of a better way?

Reply:
> If you can confirm that signal (receive) peer=crun, does not cause an error when the crun profile is missing, I think we can just leave it as is and call it for a day.

You can call it a day ;-) - the rule will just sit there and do nothing if the `crun` profile doesn't exist.